ray-operator/controllers/ray/common/pod.go

```diff
@@ -219,7 +219,7 @@ func DefaultWorkerPodTemplate(ctx context.Context, instance rayv1.RayCluster, wo
 				// Therefore, hard-coding the resources is acceptable.
 				Limits: corev1.ResourceList{
 					corev1.ResourceCPU:    resource.MustParse("200m"),
-					corev1.ResourceMemory: resource.MustParse("256Mi"),
+					corev1.ResourceMemory: resource.MustParse("512Mi"),
```

Comment:
We also need to change the memory request to match this value; otherwise, the Pod will not have a chance to be in the Guaranteed QoS class.

The Guaranteed QoS class will be assigned a lower `oom_score_adj` in cgroup that makes processes in the Pod less likely to be OOM killed. On the other hand, just scaling up the memory limit alone could higher the chance of being OOM killed.

Reply:
That makes sense, I've adapted it accordingly.